### PR TITLE
Updates for readme, licenses and pull-request template

### PR DESCRIPTION
**Pull request by marabooy**
Contributes fixes for #35 

This Pr contains 

- [x] Updates to the readme to add sections on building and how to contribute
- [x] Includes a contribution.md highlighting steps for users on contributing
- [x] Adds a PR template
 - [x] Updates licenses so that github can read and showcase the license strip

### CODE REVIEW

File: CONTRIBUTING.MD [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# How to contribute
+
+There are many ways for you to contribute to ODataConnected service The easiest way is to participate in discussion of features and issues. You can also contribute by sending pull requests of features or bug fixes to us. 
+
+## 1. Discussion
+
+You can participate into discussions and ask questions about OData .NET at our [Github issues](https://github.com/OData/ODataConnectedService/issues).
```
Comment (KenitoInc) [inline CONTRIBUTING.MD]: Replace OData.NET with OData Connected Service

File: README.md [hunk 2]
```diff
@@ -21,11 +46,18 @@ This is the official OData Connected Service repository.
...
-#### Unit Tests
 
-The `ODataConnectedService.Tests.sln` contains the unit tests for the OData connected service. The folder structure of the tests mirrors the folder structre
-of the `ODataConnectedService` project. The base namespace for tests is `Microsoft.OData.ConnectedService.Tests`. Furthermore each test class has the same name
-as the class it is testing, followed by the suffix `Test`.
+### Contribution
 
-Example: for some class `Microsoft.OData.ConnectedService.X.Y` located in `src\X\Y.cs`,
-the test class would be `Microsoft.OData.Tests.ConnectedService.Tests.X.YTest` located in `test\X\YTest.cs`
+There are many ways for you to contribute to OData .NET. The easiest way is to participate in discussion of features and issues. You can also contribute by sending pull requests of features or bug fixes to us. Contribution to the documentations is also highly welcomed. Please refer to the [CONTRIBUTING.md](https://github.com/OData/ODataConnectedService/CONTRIBUTING.md) for more details.
```
Comment (KenitoInc) [inline README.md]: Replace OData.NET with OData Connected Service

File: README.md [hunk 3]
```diff
@@ -21,11 +46,18 @@ This is the official OData Connected Service repository.
...
+
+###  Support
+
+- Issues: Report issues on [Github issues](https://github.com/OData/ODataConnectedService/issues).
+- Questions: Ask questions on [Stack Overflow](http://stackoverflow.com/questions/ask?tags=odata).
+- Feedback: Please send mails to [odatafeedback@microsoft.com](mailto:odatafeedback@microsoft.com).
+- Team blog: Please visit [http://blogs.msdn.com/b/odatateam/](http://blogs.msdn.com/b/odatateam/) and [http://www.odata.org/blog/](http://www.odata.org/blog/).
```
Comment (KenitoInc) [inline README.md]: The blogs.msdn.* link opens archived resources. Replace it with https://docs.microsoft.com/odata

File: src/ODataConnectedService.csproj.bak [hunk 4]
```diff
@@ -87,17 +87,17 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\external\Binaries\V3\Microsoft.Data.Services.Design.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.OData.Client, Version=6.17.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\packages\Microsoft.OData.Client.6.17.0\lib\net40\Microsoft.OData.Client.dll</HintPath>
+    <Reference Include="Microsoft.OData.Client, Version=7.6.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (KenitoInc) [inline src/ODataConnectedService.csproj.bak]: I don't think we need to commit the .bak file

File: CONTRIBUTING.MD (hunk 1 above)
Comment (KenitoInc) [reply CONTRIBUTING.MD]: #Resolved

File: README.md (hunk 2 above)
Comment (KenitoInc) [reply README.md]: #Resolved

File: src/ODataConnectedService.csproj.bak (hunk 4 above)
Comment (KenitoInc) [reply src/ODataConnectedService.csproj.bak]: #Resolved

File: README.md (hunk 3 above)
Comment (KenitoInc) [reply README.md]: #Resolved

File: CONTRIBUTING.MD [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+### Prepare Tools
+
+[Atom](https://atom.io/) with package [atom-beautify](https://atom.io/packages/atom-beautify) and [markdown-toc](https://atom.io/packages/markdown-toc) is recommended to edit the document. [MarkdownPad](http://www.markdownpad.com/) can also be used to edit the document.\
+Visual Studio 2017 or 2019 are the recommended tools for code contribution.
+
+### Steps to create a pull request
+
+These are the recommended steps to create a pull request:
+
+1. Create a forked repository of [(https://github.com/OData/ODataConnectedService.git](https://github.com/OData/ODataConnectedService.git)
```
Comment (odero) [inline CONTRIBUTING.MD]:
```suggestion
1. Create a forked repository of [ODataConnectedService](https://github.com/OData/ODataConnectedService.git)
```

File: CONTRIBUTING.MD [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+2. Clone the forked repository into your local environment
+3. Add a git remote to upstream for local repository with command _git remote add upstream (https://github.com/OData/ODataConnectedService.git]((https://github.com/OData/ODataConnectedService.git)_
```
Comment (odero) [inline CONTRIBUTING.MD]:
```suggestion
3. Add a git remote to upstream for local repository with command `git remote add upstream https://github.com/OData/ODataConnectedService.git`
```

File: CONTRIBUTING.MD [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+4. Make code changes and add test cases
+5. Test the changed code locally to make sure they pass
+6. Commit changed code to local repository with clear message
+7. Rebase the code to upstream via command _git pull --rebase upstream master_ and resolve conflicts if there is any then continue rebase via command _git pull --rebase continue_
```
Comment (odero) [inline CONTRIBUTING.MD]:
```suggestion
7. Rebase the code to upstream via command `git pull --rebase upstream master` and resolve conflicts if there is any then continue rebase via command `git pull --rebase continue`
```

File: CONTRIBUTING.MD [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+**Pull request is the only way we accept code and document contribution.** Pull request of document, features and bug fixes are both welcomed. Refer to this [link](https://help.github.com/articles/using-pull-requests/) to learn details about pull request. Before you send a pull request to us, you need to make sure you've followed the steps listed below.
+
+### Pick an issue to work on
+
+You should either create or pick an issue on the [issue tracker](https://github.com/OData/ODataConnectedService/issues) before you work on the pull request. After the OData team has reviewed this issue and change its label to "accepting pull request", you can work on the code change.
+
+### Prepare Tools
+
+[Atom](https://atom.io/) with package [atom-beautify](https://atom.io/packages/atom-beautify) and [markdown-toc](https://atom.io/packages/markdown-toc) is recommended to edit the document. [MarkdownPad](http://www.markdownpad.com/) can also be used to edit the document.\
```
Comment (odero) [inline CONTRIBUTING.MD]: I'd remove this para

File: CONTRIBUTING.MD [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+4. Make code changes and add test cases
+5. Test the changed code locally to make sure they pass
+6. Commit changed code to local repository with clear message
+7. Rebase the code to upstream via command _git pull --rebase upstream master_ and resolve conflicts if there is any then continue rebase via command _git pull --rebase continue_
+8. Push local commit to the forked repository
+9. Create pull request from forked repository Web console via comparing with upstream and add the ready for review label.
+10. Complete a Contributor License Agreement (CLA), refer below section for more details.
+11. Pull request will be reviewed by Microsoft OData team
+12. Address comments and revise code if necessary
+13. Commit the changes to local repository or amend existing commit via command _git commit --amend_
```
Comment (odero) [inline CONTRIBUTING.MD]:
```suggestion
13. Commit the changes to local repository or amend existing commit via command `git commit --amend`
```

File: CONTRIBUTING.MD [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+14. Rebase the code with upstream again via command _git pull --rebase upstream master_ and resolve conflicts if there is any then continue rebase via command _git pull --rebase continue_
```
Comment (odero) [inline CONTRIBUTING.MD]:
```suggestion
14. Rebase the code with upstream again via command `git pull --rebase upstream master` and resolve conflicts if there is any then continue rebase via command `git pull --rebase continue`
```

File: CONTRIBUTING.MD [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+15. Test the changed code again
+16. Push changes to the forked repository and use _--force_ option if existing commit is amended
```
Comment (odero) [inline CONTRIBUTING.MD]:
```suggestion
16. Push changes to the forked repository and use `--force` option if existing commit is amended
```

File: src/License.txt [hunk 12]
```diff
@@ -1,13 +1,23 @@
-OData Connected Service ver. 0.3.0
-
+MIT License
 Copyright (c) Microsoft Corporation
 
-All rights reserved. 
-
-MIT License
+ODataConnected service
```
Comment (odero) [inline src/License.txt]:
```suggestion
OData Connected Service
```

**Pull request merged**